File: nqueen.cpp
#include <bits/stdc++.h>
using namespace std;

bool isValid(vector<vector<int>> &grid, int row, int col){
    int i = col;
    while (i >= 0){
        if (grid[row][i] == 1){
            return false;
        }
        i--;
    }
    i = row;
    int j = col;
    while (i >= 0 && j >= 0){
        if (grid[i][j] == 1){
            return false;
        }
        i--;
        j--;
    }
    i = row;
    j = col;
    while (i < grid.size() && j >= 0){
        if (grid[i][j] == 1){
            return false;
        }
        i++;
        j--;
    }
    return true;
}

bool placeQueens(vector<vector<int>> &grid, int n){
    if (n >= grid.size())
        return true;
    for (int i = 0; i < grid.size(); i++){
        if (isValid(grid, i, n)){
            grid[i][n] = 1;
            if (placeQueens(grid, n + 1)){
                return true;
            }
            grid[i][n] = 0;
        }
    }
    return false;
}

void display(vector<vector<int>> &grid){
    for (int i = 0; i < grid.size(); i++){
        for (int j = 0; j < grid.size(); j++){
            if (grid[i][j]){
                cout << "Q ";
            }
            else{
                cout << ". ";
            }
        }
        cout << endl;
    }
}

int main(){
    int n;
    cout << "Enter the value of n : ";
    cin >> n;

    vector<vector<int>> grid(n, vector<int>(n, 0));
    placeQueens(grid, 0);
    display(grid);
    return 0;
}

File: nqueen.py
def is_valid(grid, row, col):
    for i in range(col):
        if grid[row][i] == 1:
            return False
    for i, j in zip(range(row, -1, -1), range(col, -1, -1)):
        if grid[i][j] == 1:
            return False
    for i, j in zip(range(row, len(grid), 1), range(col, -1, -1)):
        if grid[i][j] == 1:
            return False
    return True

def place_queens(grid, col):
    if col >= len(grid):
        return True
    for i in range(len(grid)):
        if is_valid(grid, i, col):
            grid[i][col] = 1
            if place_queens(grid, col + 1):
                return True
            grid[i][col] = 0
    return False

def display(grid):
    for row in grid:
        for cell in row:
            if cell:
                print("Q", end=" ")
            else:
                print(".", end=" ")
        print()

if __name__ == "__main__":
    n = int(input("Enter the value of n: "))
    grid = [[0] * n for _ in range(n)]
    if place_queens(grid, 0):
        display(grid)
    else:
        print("No solution exists.")
